velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -922,6 +922,19 @@ struct DateTruncFunction : public TimestampWithTimezoneSupport<T> {
     }
   }
 
+  FOLLY_ALWAYS_INLINE Timestamp adjustByArithmetic(
+      Timestamp timestamp,
+      long intervalSeconds,
+      const date::time_zone* timezone) {
+    long seconds = getSeconds(timestamp, timezone);
+    long s = seconds / intervalSeconds;
+    if (seconds < 0) {
+      s = s - 1;
```

Comment:
Do we need to minus 1 if `seconds % intervalSeconds == 0`?

Reply:
Good point, negative timestamp really complicate the issue.`date_trunc('second', timestamp(-1, 0))` should be `timestamp(-1, 0)`.
And it makes me think that the current function does not take nanoseconds into account.
eg. 
- `date_trunc('second', timestamp(1, -1))`
- `date_trunc('second', timestamp(-1, 1))` 
- `date_trunc('second', timestamp(-1, -1))`

Reply:
Sorry, I did't notice that nanos is an uint64_t. So we can never create `timestamp(1, -1)`. And a positive nanos will always be truncated.

Reply:
Thanks for the fix. Would you like to add a test for `seconds % intervalSeconds == 0`?